### PR TITLE
[MFTF] User sees images attributes in the image preview

### PR DESCRIPTION
**Pull request by engcom-Golf**
### Description (*)
Implemented MFTF test to cover scenario User sees images attributes in the image preview

### Fixed Issues (if relevant)
magento/adobe-stock-integration#360 :User sees images attributes in the image preview

### Manual testing scenarios (*)
Run Functional tests

### CODE REVIEW

**Review by diazwatson — DISMISSED** [review]

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 <sections xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/SectionObject.xsd">
     <section name="AdobeStockSection">
-        <element name="slideOutPanelButton" type="button" selector="[data-ui-id=wysiwyg-images-content-search-adobe-stock-button]"/>
+        <element name="slideOutPanelButton" type="button" selector="//*[@data-ui-id='wysiwyg-images-content-search-adobe-stock-button']"/>
```
Comment (hws47a) [inline AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml]: What is the reason for changing this selector?

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminUserSeesImagesAttributesInTheImagePreviewTest.xml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        </before>
+        <after>
+            <actionGroup ref="AdminAdobeStockSetConfigActionGroup" stepKey="cleanConfig">
+                <argument name="enabled" value="0"/>
+                <argument name="wysiwygEnabled" value="disabled"/>
+<!--                TODO: uncomment next lines where issue with sensitive config setting will be resolved. -->
+<!--                <argument name="apiKey" value=""/>-->
+<!--                <argument name="privateKey" value=""/>-->
+            </actionGroup>
+            <amOnPage url="admin/admin/auth/logout/" stepKey="amOnLogoutPage"/>
```
Comment (hws47a) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminUserSeesImagesAttributesInTheImagePreviewTest.xml]: There is a specific action group for logout action
`<actionGroup ref="logout" stepKey="logout"/>`

**Review by hws47a — DISMISSED** [review]

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml [hunk 3]
```diff
@@ -23,5 +23,7 @@
         <element name="savePreview" type="block" selector="//button[@class='action-secondary']/span[contains(.,'Save Preview')]"/>
         <element name="imagePreview" type="block" selector="//div[@class='masonry-image-preview']//img"/>
         <element name="imagePreviewNavigation" type="block" selector="//div[@class='masonry-image-preview']//div[@class='action-buttons']/button[@class='action-{{type}}']" parameterized="true"/>
+        <element name="imageAttribute" type="block" selector="//*[@id='adobe-stock-images-search-modal']//div[@class='attributes']//span[text()='{{type}}']/parent::div//div[@class='value']//span" parameterized="true"/>
+        <element name="imageAttributeTitle" type="block" selector="//*[@id='adobe-stock-images-search-modal']//div[@class='attributes']//span[text()='{{title}}']" parameterized="true"/>
```
Comment (hws47a) [inline AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml]: My personal opinion about how selectors can be improved:
1. I don't think that @class attribute is reliable (it has a styling purpose) and should be used for selecting items in tests. I would go for data attributes (in a perfect scenario with a clear testing purpose) instead.
2. There is duplication for a modal selector. I think it's better to reuse "modal" in a way I did here: https://github.com/magento/adobe-stock-integration/blob/develop/AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminAssertAdobeStockThumbnailsNumberActionGroup.xml

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml [hunk 4]
```diff
@@ -23,5 +23,7 @@
         <element name="savePreview" type="block" selector="//button[@class='action-secondary']/span[contains(.,'Save Preview')]"/>
         <element name="imagePreview" type="block" selector="//div[@class='masonry-image-preview']//img"/>
         <element name="imagePreviewNavigation" type="block" selector="//div[@class='masonry-image-preview']//div[@class='action-buttons']/button[@class='action-{{type}}']" parameterized="true"/>
+        <element name="imageAttribute" type="block" selector="//*[@id='adobe-stock-images-search-modal']//div[@class='attributes']//span[text()='{{type}}']/parent::div//div[@class='value']//span" parameterized="true"/>
```
Comment (hws47a) [inline AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml]: Just a question, why do you prefer XPath selectors instead of CSS selectors?

Comment (engcom-Golf) [reply AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml]: Hi @hws47a thank you for your review, for me it's easier to debug or find the element. for example to find element i'm use this` $x("//small/parent::*[@class='filecnt selected']//img")` and i think for perfomance it's better to use xpath,  but as devdoc say's this is not good practice, so i agree with you and change this according to devdocs.
DeepinScreenshot_select-area_20190827093531

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml (hunk 3 above)
Comment (engcom-Golf) [reply AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml]: ok

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminUserSeesImagesAttributesInTheImagePreviewTest.xml (hunk 2 above)
Comment (engcom-Golf) [reply AdobeStockImageAdminUi/Test/Mftf/Test/AdminUserSeesImagesAttributesInTheImagePreviewTest.xml]: ok

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml (hunk 1 above)
Comment (engcom-Golf) [reply AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml]: Seems like i'm did this  for mistake, i revert it back. thanks.

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml (hunk 3 above)
Comment (engcom-Golf) [reply AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml]: 2. If i have `parameterized="true"` then i not have to use this approach, is this expected be as parameter

**Comment by engcom-Golf** [comment]
Hi @hws47a thank you for your comment, can you please review new changes ?

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml (hunk 3 above)
Comment (sivaschenko) [reply AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSection.xml]: Let's refactor all the selectors including the modal selector in a separate pull request
`//*[@id='adobe-stock-images-search-modal']//div[@data-role='image-attributes-value']//span[text()='{{title}}']` -> `div[data-role='image-attributes-value'] span:contains('{{title}}')`

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**